questionnaire.js

```diff
@@ -361,11 +361,18 @@ export function isFirstQuestion() {
   return questionQueue.isEmpty() || questionQueue.isFirst();
 }
 
-function numberOfInputs(element) {
-  let resps = Array.from(
-    element.querySelectorAll("input, textarea, select")
-  ).reduce((acc, current) => {
-    //if (["submit", "button"].includes(current.type)) return acc;
+/**
+ * Determine the storage format for the response data.
+ * Grid questions and questions with multiple response inputs are stored as objects. Ensure each key is stored with the response.
+ * Single response (radio) input questions are stored as primitives.
+ * Multi-selection (checkbox) input questions are stored as arrays.
```

Comment:
so to confirm... checkboxes are arrays stored within an object?

Reply:
I cleaned up this JSDoc in the most recent push, to make it more clear:

Primitives: key: "value"
<img width="189" alt="Screenshot 2024-07-26 at 9 08 55 AM" src="https://github.com/user-attachments/assets/831e484d-8037-496c-b72e-8ec06acd9c06">


Arrays: key: ["value1", "value2", "value3", ...etc]
<img width="197" alt="Screenshot 2024-07-26 at 8 55 16 AM" src="https://github.com/user-attachments/assets/85c017db-b2fb-47bc-8f54-2db9a4a14f0a">


Objects: parentKey: { key1: "value1", key2: "value2", etc... }
<img width="302" alt="Screenshot 2024-07-26 at 9 10 41 AM" src="https://github.com/user-attachments/assets/20e373a8-98ba-43a4-808e-8205878a4a3e">